src/core/server/saved_objects/migrations/kibana/kibana_migrator.ts

```diff
@@ -163,6 +164,15 @@ export class KibanaMigrator {
       registry: this.typeRegistry,
     });
 
+    this.log.debug('Applying registered migrations for the following saved object types:');
+    Object.entries(this.documentMigrator.migrationVersion)
+      .sort(([t1, v1], [t2, v2]) => {
+        return Semver.compare(v1, v2);
```

Comment:
NIT: are `v1` and `v2` ensured to always exist? If any don't, can this logic break the rest of the migration logic?

Reply:
The code is hard to follow, but from what I understand, I think that `this.documentMigrator.migrationVersion` will always have a version, but that types without any migration registered will not be present in the map. Which sounds acceptable @rudolf?

https://github.com/elastic/kibana/blob/8534faf7eeb090a59c3edb5f917cdb89ae7ff179/src/core/server/saved_objects/migrations/core/document_migrator.ts#L388-L390

https://github.com/elastic/kibana/blob/8534faf7eeb090a59c3edb5f917cdb89ae7ff179/src/core/server/saved_objects/migrations/core/document_migrator.ts#L178-L189

Reply:
yeah, these versions come from plugins registering migrations, so it will always be present (and I think we validate that it's a valid semver). If a type doesn't register a migration it won't be in this map.